llm/gguf.go

```diff
@@ -87,38 +86,41 @@ func (llm *ggufModel) NumKV() uint64 {
 	return llm.V2.NumKV
 }
 
-func (llm *ggufModel) ModelFamily() ModelFamily {
+func (llm *ggufModel) ModelFamily() string {
 	t, ok := llm.kv["general.architecture"].(string)
 	if ok {
-		return ModelFamily(t)
+		return t
 	}
 
-	log.Printf("unknown model family: %T", t)
-	return ModelFamilyUnknown
+	return "unknown"
 }
 
-func (llm *ggufModel) ModelType() ModelType {
+func (llm *ggufModel) ModelType() string {
 	switch llm.ModelFamily() {
-	case ModelFamilyLlama:
-		blocks, ok := llm.kv["llama.block_count"].(uint32)
-		if ok {
-			return ModelType(blocks)
+	case "llama":
+		if blocks, ok := llm.kv["llama.block_count"].(uint32); ok {
+			if numGQA, ok := llm.kv["llama.num_gqa"].(uint32); ok && numGQA == 0 {
```

Comment:
what about grouped query attention being 0 indicates that the model is 70B?

Reply:
That was a typo. It should be 8. The reason is 65B of llama and 70B of llama2 both have 80 layers. However 70B llama2 has grouped query attention so we can use that differentiate between the two